pkg/kubelet/kuberuntime/util/util_test.go

```diff
@@ -0,0 +1,227 @@
+				SandboxStatuses: []*runtimeapi.PodSandboxStatus{
+					{
+						Id:       "sandboxID2",
+						Metadata: &runtimeapi.PodSandboxMetadata{Attempt: uint32(1)},
+						State:    runtimeapi.PodSandboxState_SANDBOX_READY,
+					},
+					{
+						Id:       "sandboxID1",
+						Metadata: &runtimeapi.PodSandboxMetadata{Attempt: uint32(0)},
+						State:    runtimeapi.PodSandboxState_SANDBOX_READY,
+					},
+				},
+			},
+			true,
+			2,
+			"sandboxID2",
+		},
+		"Pod with no ready sandbox statuses": {
+			&v1.Pod{},
+			&kubecontainer.PodStatus{
+				SandboxStatuses: []*runtimeapi.PodSandboxStatus{
+					{
+						Id:       "sandboxID2",
+						Metadata: &runtimeapi.PodSandboxMetadata{Attempt: uint32(1)},
+						State:    runtimeapi.PodSandboxState_SANDBOX_NOTREADY,
+					},
+					{
+						Id:       "sandboxID1",
+						Metadata: &runtimeapi.PodSandboxMetadata{Attempt: uint32(0)},
+						State:    runtimeapi.PodSandboxState_SANDBOX_NOTREADY,
+					},
+				},
+			},
+			true,
+			2,
+			"sandboxID2",
+		},
+		"Pod with ready sandbox status but network namespace mismatch": {
+			&v1.Pod{
+				Spec: v1.PodSpec{
+					HostNetwork: true,
+				},
+			},
+			&kubecontainer.PodStatus{
+				SandboxStatuses: []*runtimeapi.PodSandboxStatus{
+					{
+						Id: "sandboxID1",
+						Linux: &runtimeapi.LinuxPodSandboxStatus{
+							Namespaces: &runtimeapi.Namespace{
+								Options: &runtimeapi.NamespaceOption{
+									Network: runtimeapi.NamespaceMode_POD,
+								},
+							},
+						},
+						Metadata: &runtimeapi.PodSandboxMetadata{Attempt: uint32(0)},
+						State:    runtimeapi.PodSandboxState_SANDBOX_READY,
+					},
+				},
+			},
+			true,
+			1,
+			"",
+		},
+		"Pod with ready sandbox status but no IP": {
+			&v1.Pod{
+				Spec: v1.PodSpec{
+					HostNetwork: false,
+				},
+			},
+			&kubecontainer.PodStatus{
+				SandboxStatuses: []*runtimeapi.PodSandboxStatus{
+					{
+						Id: "sandboxID1",
+						Network: &runtimeapi.PodSandboxNetworkStatus{
+							Ip: "",
+						},
+						Metadata: &runtimeapi.PodSandboxMetadata{Attempt: uint32(0)},
+						State:    runtimeapi.PodSandboxState_SANDBOX_READY,
+					},
+				},
+			},
+			true,
+			1,
+			"sandboxID1",
+		},
+		"Pod with ready sandbox status with IP": {
+			&v1.Pod{
+				Spec: v1.PodSpec{
+					HostNetwork: false,
+				},
+			},
+			&kubecontainer.PodStatus{
+				SandboxStatuses: []*runtimeapi.PodSandboxStatus{
+					{
+						Id: "sandboxID1",
+						Network: &runtimeapi.PodSandboxNetworkStatus{
+							Ip: "10.0.0.10",
+						},
+						Metadata: &runtimeapi.PodSandboxMetadata{Attempt: uint32(0)},
+						State:    runtimeapi.PodSandboxState_SANDBOX_READY,
+					},
+				},
+			},
+			false,
+			0,
+			"sandboxID1",
+		},
+	} {
+		t.Logf("TestCase: %s", desc)
+		changed, attempt, id := PodSandboxChanged(test.input, test.status)
+		assert.Equal(t, test.expectedChanged, changed)
+		assert.Equal(t, test.expectedAttempt, attempt)
+		assert.Equal(t, test.expectedSandboxID, id)
```

Comment:
```suggestion
		t.Run(desc, func(t *testing.T) {
			changed, attempt, id := PodSandboxChanged(test.input, test.status)
			require.Equal(t, test.expectedChanged, changed)
			require.Equal(t, test.expectedAttempt, attempt)
			require.Equal(t, test.expectedSandboxID, id)
		})
```

prefer require over assert (ends tests vs stacking failures), and `t.Run` over logging for better output.

Reply:
fixed